FtpClient/mainwindow.cpp

```diff
@@ -51,6 +61,74 @@ void MainWindow::slotError(QAbstractSocket::SocketError err)
     }
 }
 
+void MainWindow::onChangeDir()
+{
+    sendPath("C:\\FTPclient");
+
+    socket->waitForReadyRead();
+    QByteArray receivedData = socket->readAll();
+
+    Packet pack(receivedData);
+
+    if(pack.getErrorCode())
+    {
+        QMessageBox::information(this, "ERROR", pack.getData());
+    }
+    else
+    {
+        qDebug () << pack.getData();
+    }
+}
+
+void MainWindow::onDownloadFile()
+{
+    std::string buffer;
+
+    buffer.resize(2);
+
+    *reinterpret_cast<short*>(&buffer[0]) = 1;
+
+    buffer += "Download file plz";
+
+    socket->write( buffer.c_str(), buffer.size() );
+}
+
+void MainWindow::onUploadFile()
+{
+
+    QFile readStream("data.png");
+
+    if( !readStream.open( QIODevice::ReadOnly ) )
+    {
+        qDebug() << "Can`t open file";
+        return;
+    }
+
+    qDebug() << readStream.size();
+
+    QByteArray data( 2, '0' );
+
+    *reinterpret_cast<short*>( data.data() ) = 2;
+
+    data.append( readStream.readAll() );
+
+    socket->write( data.data(), data.size() );
+}
+
+void MainWindow::sendPath(const QString &path)
+{
+    Packet pack(0, 0, 0, path);
+    if(pack.isValid())
```

Comment:
what the point checking after construction?
Implement a constructor which throws exception on failure.